drivers/ir_nec/ir_nec.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2022 Dario Petrillo
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ir_nec
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the IR NEC Remote receiver
+ *
+ * @author      Dario Petrillo <dario.pk1@gmail.com>
+ *
+ * @}
+ */
+
+#include "ir_nec.h"
+#include "ir_nec_constants.h"
+#include "ir_nec_params.h"
+
+#include <stdlib.h>
+#include "ztimer.h"
+
+#include "ir_nec.h"
+
+static void ir_nec_isr(void *arg)
+{
+    ir_nec_t *ir = (ir_nec_t *)arg;
+
+    if (gpio_read(ir->pin) != 0) { // rising edge
+        ir->last_rising = ztimer_now(ZTIMER_USEC);
+    }
+    else { // falling edge
+        int length = ztimer_now(ZTIMER_USEC) - ir->last_rising;
+
+        if (abs(length - IR_NEC_START_US) <= IR_NEC_EPS_US) {
+            ir->data = 0;
+            ir->read_bits = 0;
+        }
+        else if ((length - IR_NEC_ZERO_US) <= IR_NEC_EPS_US && ir->read_bits != 0xff) {
```

Comment:
```suggestion
        else if ((length - IR_NEC_ZERO_US) <= IR_NEC_EPS_US && ir->read_bits < 32) {
```

using `<32` this also checks the bounds of the received code